include/libdnf5-cli/output/moduleinfo.hpp

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef LIBDNF_CLI_OUTPUT_MODULEINFO_HPP
+#define LIBDNF_CLI_OUTPUT_MODULEINFO_HPP
+
+#include "key_value_table.hpp"
+#include "utils/string.hpp"
+
+#include "libdnf5-cli/tty.hpp"
+
+#include <libdnf5/module/module_item.hpp>
+#include <libdnf5/module/module_sack.hpp>
+#include <libsmartcols/libsmartcols.h>
+
+#include <set>
+#include <string>
+
+namespace libdnf5::cli::output {
+
+
+class ModuleInfo : public KeyValueTable {
+public:
+    template <typename ModuleItem>
+    void add_module_item(ModuleItem & module_item);
+
+private:
+    void add_multiline_value(const char * key, const std::vector<std::string> & multiline_value);
+};
+
+
+void ModuleInfo::add_multiline_value(const char * key, const std::vector<std::string> & values) {
```

Comment:
May I ask you why the code is not in `CPP`?

Reply:
Isn't it because it's used in the template below?

Reply:
Function `add_line()` has a code in `cpp`